server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/MirrorConfig.java

```diff
@@ -147,12 +192,22 @@ id, enabled, schedule, direction, findCredential(credentials, credentialId),
         throw new IllegalArgumentException("could not find a mirror provider for " + mirrorContext);
     }
 
-    public static Credential findCredential(Iterable<Credential> credentials,
-                                            @Nullable String credentialId) {
-        if (credentialId != null) {
-            for (Credential c : credentials) {
+    private Credential findCredential(Map<String, List<Credential>> repoCredentials,
+                                      List<Credential> projectCredentials) {
+        if (repoCredential) {
```

Comment:
Suggestion) This logic seems complicated to be included in a data class. What do you think of extracting the methods to `MirrorConverter`?